tests/sentry/backup/test_releases.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import os
+import tempfile
+from pathlib import Path
+
+import yaml
+
+from sentry.backup.comparators import get_default_comparators
+from sentry.backup.dependencies import NormalizedModelName
+from sentry.backup.imports import import_in_global_scope
+from sentry.backup.scopes import ExportScope
+from sentry.backup.validate import validate
+from sentry.testutils.helpers.backups import BackupTestCase, export_to_file
+from sentry.testutils.pytest.fixtures import read_snapshot_file
+from sentry.utils import json
+from tests.sentry.backup import mark, targets
+
+RELEASE_TESTED: set[NormalizedModelName] = set()
+
+
+class ReleaseTests(BackupTestCase):
+    """
+    Ensure that exports from the last two released versions of self-hosted are still able to be
+    imported.
+    """
+
+    @classmethod
+    def get_snapshot_path(cls, release: str) -> str:
+        root_dir = os.path.dirname(os.path.realpath(__file__))
+        return f"{root_dir}/snapshots/{cls.__name__}/test_at_{release.replace('.', '_')}.pysnap"
+
+    @targets(mark(RELEASE_TESTED, "__all__"))
+    def test_at_head(self):
+        self.create_exhaustive_instance(is_superadmin=True)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = Path(tmp_dir).joinpath(f"{self._testMethodName}.json")
+            exported_data = export_to_file(tmp_path, ExportScope.Global)
+
+            # Note: because we are using the 'insta_snapshot` feature of pysnap, the files will be
+            # saved as annotated YAML files, not JSON. While this is not strictly a supported format
+            # for relocation (it's JSON-only), since YAML is a superset of JSON, this should be
+            # okay, and we can think of these files as valid JSON exports saved using a slightly
+            # different presentation for ease of testing.
+            def inequality_comparator(refval: str, output: str) -> str | bool:
+                refval_json = yaml.safe_load(refval) or dict()
+                output_json = yaml.safe_load(output)
+                result = validate(refval_json, output_json, get_default_comparators())
+                if not result.empty():
+                    # Instead of returning a simple diff, which will differ in ways that are not
+                    # necessarily relevant to the comparison (newly minted tokens, etc), we just
+                    # return the validation comparison findings in pretty-printed form.
+                    return "The following inconsistences were found:\n\n" + result.pretty()
+
+                return False
+
+            self.insta_snapshot(exported_data, inequality_comparator=inequality_comparator)
+            return exported_data
+
+    def test_at_23_9_1(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            _, snapshot_refval = read_snapshot_file(self.get_snapshot_path("23.9.1"))
+            snapshot_data = yaml.safe_load(snapshot_refval)
```

Comment:
nit: I'm assuming most of this code will be reused in the future, is it worth abstracting that to another function? I guess we will only support 2 minor releases so it's not a big deal

Reply:
We'll have two test methods (as described in my comment on the main thread), so it will be a bit cleaner if we have a standalone function. This is especially true when we add automation to replace these test cases via codegen.